src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -169,21 +169,22 @@ public void editStats(int type) {
     }
 
     @Override
-    public ArrayList<Integer> getPerformance() {
+    public ArrayList<Float> getPerformance() {
         return performance;
     }
 
     @Override
     public void updatePerformance(Model model) {
         requireNonNull(model);
-        model.updateFilteredFlashCardList(new RatingContainsKeywordPredicate("good"));
-        int numGood = model.getFilteredFlashCardList().size();
-        model.updateFilteredFlashCardList(new RatingContainsKeywordPredicate("hard"));
-        int numHard = model.getFilteredFlashCardList().size();
-        model.updateFilteredFlashCardList(new RatingContainsKeywordPredicate("easy"));
-        int numEasy = model.getFilteredFlashCardList().size();
-
-        int value = ((numEasy + numGood) * 100) / (numEasy + numGood + numHard);
+        int numGood = model.getFilteredFlashCardListNoCommit(new RatingContainsKeywordPredicate("good")).size();
+        int numHard = model.getFilteredFlashCardListNoCommit(new RatingContainsKeywordPredicate("hard")).size();
+        int numEasy = model.getFilteredFlashCardListNoCommit(new RatingContainsKeywordPredicate("easy")).size();
+        float value;
+        if ((numEasy + numGood + numHard) == 0) {
+            value = 0;
+        } else {
+            value = ((numEasy + numGood) * 100) / (numEasy + numGood + numHard);
```

Comment:
or change this to value = ((float) numEasy + numGood) * 100 /(numEasy + numGood + numHard);